### PR TITLE
Add collections endpoints

### PR DESCRIPTION
**Pull request by geospatial-roman**
Issue: [#7](https://github.com/eodcgmbh/openeo-fastapi/issues/7)
Add `/collections` and` /collections/{collection_id}` endpoints, to query STAC API for collection metadata

Additional changes:
- post init for app to register methods
- tests and mock test app
- STAC data model definitions

### CODE REVIEW

File: openeo_fastapi/client/collections.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+async def get_collections():
+    """
+    Basic metadata for all datasets
+    """
+
+    async with aiohttp.ClientSession() as client:
+        async with client.get(os.getenv("STAC_API_URL") + "collections") as response:
+            resp = await response.json()
```
Comment (SerRichard) [inline openeo_fastapi/client/collections.py:19]: think about pydantic settings here :)

File: openeo_fastapi/client/models.py [hunk 2]
```diff
@@ -0,0 +1,604 @@
+import sys
```
Comment (geospatial-roman) [inline openeo_fastapi/client/models.py:1]: @SerRichard @GeraldIr Do we want to declare all the STAC related models here? or import them from stac-utils / stac-fastapi ?

**Review by SerRichard — COMMENTED** [review]
Lgtm, as discussed #14 will add more to this

File: openeo_fastapi/client/models.py (hunk 2 above)
Comment (SerRichard) [reply openeo_fastapi/client/models.py:1]: Just took a quick look, unless I'm missing something, they don't seem to have a number of the models? But, if I've missed something and a bunch of these can be imported and not defined here, sure thing!

Comment (geospatial-roman) [reply openeo_fastapi/client/models.py:1]: ok, will import what has already been definded in stac-utils and keep the rest

**Review by SerRichard — APPROVED** [review]
lgtm

**Pull request merged**